### PR TITLE
Multi metric support for ab tests (using redis)

### PR DESCRIPTION
**Pull request by sdhull**
This is really a WIP PR because I only implemented the support for redis adapter. I have neither the time nor the inclination to get this working for mongo & AR, though I was careful not to _break_ those adapters.

You can feel free to drag this across the finish line, as I am allowing edits from vanity maintainers. This also may be useful for others who are struggling with similar problems (eg, tracking funnel conversion with experiments).

Improvements:
- track each metric conversion separately for a given ab test [redis only]
- show conversions per metric on the dashboard [redis only]
- show unique & "total" conversion counters on dashboard ("total" previously completely hidden & unused afaict)
- make the dashboard full width
- fixes "total" conversion tracking to respect the `implicit` param (eg, if you are tracking "dollars added to cart", this is being tracked in current vanity release regardless of whether or not that conversion was done by a participant in your experiment)

### CODE REVIEW

**Comment by phillbaker** [comment]
@sdhull thanks for the PR, it'll take me a couple of days to review, but this is awesome!

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@ html
 .rvmrc
 .ruby-version
 .ruby-gemset
+tags
```
Comment (phillbaker) [inline .gitignore]: Hm, if you're generating ctags via an editor plugin, what do you think about putting this in a global gitignore in your home directory instead of per project?

File: lib/vanity/adapters/abstract_adapter.rb [hunk 2]
```diff
@@ -48,72 +48,81 @@ def destroy_metric(metric)
 
       # -- Experiments --
 
-      def experiment_persisted?(experiment)
+      def experiment_persisted?(experiment_id)
```
Comment (phillbaker) [inline lib/vanity/adapters/abstract_adapter.rb]: Since this would be a breaking, non-backward compatible change, can we pull this change into a separate PR? When these changes are made, we also should be included backward compatible code, for example, see https://github.com/assaf/vanity/pull/317. What do you think about keeping these methods accepting `experiment` for this PR?

File: lib/vanity/templates/vanity.css [hunk 3]
```diff
@@ -1,3 +1,5 @@
+.vanity { width: 90%; margin: 2em auto; font-family: "Helvetica Neue", "Helvetica", "Verdana", sans-serif }
```
Comment (phillbaker) [inline lib/vanity/templates/vanity.css]: This change to almost-full width and other style changes would also be great to pull into a separate PR.

File: lib/vanity/adapters/abstract_adapter.rb [hunk 4]
```diff
@@ -48,72 +48,81 @@ def destroy_metric(metric)
...
       def ab_counts(experiment, alternative)
         fail "Not implemented"
       end
 
+      # Returns counts by metric for a given A/B experiment (actual object) & alternative (by index).
+      # Returns a hash with keys metric id and values that are hashes like {added_to_cart: {converted: x, conversions: y}}
+      # Keep in mind that "converted" is unique conversions while "conversions" is a total tally (or
+      # you could set it up to track dollar sales or something like that)
+      # only really supported for redis adapter
+      def ab_counts_by_metric(experiment, alternative)
```
Comment (phillbaker) [inline lib/vanity/adapters/abstract_adapter.rb]: Can you add this new method to `doc/adapters.textile`? We wouldn't need to include which adapters support it there.

File: lib/vanity/adapters/abstract_adapter.rb [hunk 5]
```diff
@@ -122,23 +131,23 @@ def ab_assigned(experiment, identity)
       # true, add particpant if not already recorded for this experiment. If
       # implicit is false (default), only add conversion is participant
       # previously recorded as participating in this experiment.
-      def ab_add_conversion(experiment, alternative, identity, count = 1, implicit = false)
+      def ab_add_conversion(experiment_id, alternative, identity, options={})
```
Comment (phillbaker) [inline lib/vanity/adapters/abstract_adapter.rb]: Makes sense to condense these into a `options` parameter, but we'll probably have to make this change in a backwards compatible manner as well, see https://github.com/assaf/vanity/pull/317 for an example of that.

File: .gitignore (hunk 1 above)
Comment (sdhull) [inline .gitignore]: Dang sad I only just saw this suggestion -- it's brilliant

**Comment by bensheldon** [comment]
Closing this for inactivity. Please comment/reopen if you'd like to pick it back up.
